src/include/migraphx/shape.hpp

```diff
@@ -431,6 +431,30 @@ struct MIGRAPHX_EXPORT shape
     std::shared_ptr<const shape_impl> impl;
 };
 
+// "Almost identical" shapes.  To support an MLIR feature, there is a limited
+// case where shapes may both be standard but have non-identical strides.
+static bool inline is_compatible_shape(const shape& actual, const shape& expected)
+{
+    // Check subshapes
+    if(expected.type() == shape::tuple_type)
+        return equal(actual.sub_shapes().begin(),
+                     actual.sub_shapes().end(),
+                     expected.sub_shapes().begin(),
+                     &is_compatible_shape);
+    // Only the expected can be dynamic
+    if(expected.dynamic())
+        return true;
+    if(actual == expected)
+        return true;
+    if(actual.type() != expected.type())
+        return false;
+    // If both shapes are standard and lens match, they are considered compatible
+    // even if strides are different.
+    if(actual.standard() and expected.standard())
+        return actual.lens() == expected.lens();
+    return false;
+}
```

Comment:
THis should just go into the program.cpp file.

Reply:
done, part of the above request.